src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp

```diff
@@ -640,4 +641,111 @@ TEST_CASE("RotatedSPOs construct delta matrix", "[wavefunction]")
     CHECK(new_params2[i] == Approx(old_params[i]));
 }
 
+namespace testing
+{
+opt_variables_type& getMyVars(SPOSet& rot) { return rot.myVars; }
+opt_variables_type& getMyVarsFull(RotatedSPOs& rot) { return rot.myVarsFull; }
+std::vector<std::vector<QMCTraits::RealType>>& getHistoryParams(RotatedSPOs& rot) { return rot.history_params_; }
+} // namespace testing
+
+// Test using global rotation
+TEST_CASE("RotatedSPOs read and write parameters", "[wavefunction]")
+{
+  auto fake_spo = std::make_unique<FakeSPO>();
+  fake_spo->setOrbitalSetSize(4);
+  RotatedSPOs rot("fake_rot", std::move(fake_spo));
+  int nel = 2;
+  rot.buildOptVariables(nel);
+
+  optimize::VariableSet vs;
+  rot.checkInVariablesExclusive(vs);
+  vs[0] = 0.1;
+  vs[1] = 0.15;
+  vs[2] = 0.2;
+  vs[3] = 0.25;
+  rot.resetParametersExclusive(vs);
+
+  hdf_archive hout;
+  vs.writeToHDF("rot_vp.h5", hout);
+
+  rot.writeVariationalParameters(hout);
+
+  hout.close();
```

Comment:
Maybe introducing a scope for hout and let the destructor call close()? The code can be more readable.